compiler/src/dotty/tools/dotc/typer/ProtoTypes.scala

```diff
@@ -500,29 +500,6 @@ object ProtoTypes {
           // We only need to propagate constraints if we typed the arguments in a different
           // TyperState and if that created additional constraints.
           if (passedTyperState ne protoTyperState) && (oldConstraint ne newConstraint) then
-            // To respect the pre-condition of `mergeConstraintWith` and keep
-            // `protoTyperState` committable we must ensure that it does not
-            // contain any type variable which don't already exist in the passed
-            // TyperState. This is achieved by instantiating any such type
-            // variable. NOTE: this does not suffice to discard type variables
-            // in ancestors of `protoTyperState`, if this situation ever
-            // comes up, an assertion in TyperState will trigger and this code
-            // will need to be generalized.
```

Comment:
I believe this code is flawed since the situation in the NOTE can happen which I think is what causes https://github.com/scala/scala3/issues/21555 and similar issues, but it's not clear to me that just removing this is fine, and it seems unrelated to the rest of this PR?

Reply:
I went to reverify things, and I can't find a test case that fails without your code here (and without any thing in this PR).  So I can't say that my code replaces it, because it doesn't seem to do anything for any of our tests.  So I put it back (by dropping my commit).